src/interactive/scala/tools/nsc/interactive/Global.scala

```diff
@@ -1139,23 +1139,21 @@ class Global(settings: Settings, _reporter: Reporter, projectName: String = "")
     //print("add members")
     for (sym <- ownerTpe.members)
       addTypeMember(sym, pre, sym.owner != ownerTpe.typeSymbol, NoSymbol)
-    members.allMembers #:: {
-      //print("\nadd enrichment")
-      val applicableViews: List[SearchResult] =
-        if (ownerTpe.isErroneous) List()
-        else new ImplicitSearch(
-          tree, functionType(List(ownerTpe), AnyTpe), isView = true, isByNamePt = false,
-          context0 = context.makeImplicit(reportAmbiguousErrors = false)).allImplicits
-      for (view <- applicableViews) {
-        val vtree = viewApply(view)
-        val vpre = stabilizedType(vtree)
-        for (sym <- vtree.tpe.members if sym.isTerm) {
-          addTypeMember(sym, vpre, inherited = false, view.tree.symbol)
-        }
+    //print("\nadd enrichment")
+    // members.clear()
+    val applicableViews: List[SearchResult] =
+      if (ownerTpe.isErroneous) List()
+      else new ImplicitSearch(
+        tree, functionType(List(ownerTpe), AnyTpe), isView = true, isByNamePt = false,
+        context0 = context.makeImplicit(reportAmbiguousErrors = false)).allImplicits
+    for (view <- applicableViews) {
+      val vtree = viewApply(view)
+      val vpre = stabilizedType(vtree)
+      for (sym <- vtree.tpe.members if sym.isTerm) {
+        addTypeMember(sym, vpre, inherited = false, view.tree.symbol)
       }
-      //println()
-      LazyList(members.allMembers)
     }
+    LazyList(members.allMembers)
```

Comment:
It seems the idea of the old code was to return a `LazyList(members, extensionMethods)` where the `extensionMethods` were computed lazily only if needed.

I see `typeMembers` is used in two places, one is `typeMembers(qualPos).toList.flatten` so everythings is forced immediately. This one is used from `nsc.interpreter.PresentationCompileResult`, which seems to be what the REPL is using.

The other is `respondGradually(response) { typeMembers(pos) }`, which is used from `nsc.interactive.CompilerControl.askTypeCompletion`, so that's for IDEs. It seems the goal there is to show direct members in the IDE fast without having to wait for implicit search to complete. I'm not sure if there is any use case for this remaining. @retronym wdyt?

Funny discovery on the side: what on earth is this? The `complete` command calls `CompilerControl.askTypeCompletion`.

```
$> scala scala.tools.nsc.interactive.REPL
> help
Available commands:
 reload <file_1> ... <file_n>
 reloadAndAskType <file> <sleep-ms>
 typed <file>
 typeat <file> <start-pos> <end-pos>
 typeat <file> <pos>
 complete <file> <start-pos> <end-pos>
 compile <file> <pos>
 structure <file>
 quit
```

Reply:
For comparison, your version is here: 1450caf43be621a0b2e6e300489d7c15bd4d97e8

I'm looking at whether I can fix this in a way that doesn't regress but also doesn't lose the lazy aspect.